examples/seismic/source.py

```diff
@@ -93,9 +93,11 @@ def __new__(cls, **kwargs):
         time_order = kwargs.pop('time_order', 2)
         p_dim = kwargs.pop('dimension', Dimension(name='p_%s' % name))
 
-        # Either `npoint` or `coordinates` must be provided
         coordinates = kwargs.pop('coordinates', None)
-        npoint = kwargs.pop('npoint', None) or coordinates.shape[0]
+        # Either `npoint` or `coordinates` must be provided
+        npoint = kwargs.pop('npoint', None)
+        if npoint is None:
+            npoint = coordinates.shape[0]
```

Comment:
Let's take the opportunity to raise a good exception here?  Like "npoint or coordinates must be provided"?  The current error will be "object of type NoneType doesn't have attribute shape" or something similar.